libnmstate/nm/profile.py

```diff
@@ -21,14 +21,948 @@
+
+            for dev in devs_to_delete.values():
+                device.delete_device(context, dev)
+            self._ctx.wait_all_finish()
+
+    def _get_affected_devices(self):
+        nmdev = self.context.get_nm_dev(self.iface_state[Interface.NAME])
+        devs = []
+        if nmdev:
+            devs += [nmdev]
+            iface_type = self.iface_state[Interface.TYPE]
+            if iface_type == InterfaceType.OVS_BRIDGE:
+                port_slaves = ovs.get_slaves(nmdev)
+                iface_slaves = [
+                    iface
+                    for port in port_slaves
+                    for iface in ovs.get_slaves(port)
+                ]
+                devs += port_slaves + iface_slaves
+            elif iface_type == LB.TYPE:
+                devs += bridge.get_slaves(nmdev)
+            elif iface_type == bond.BOND_TYPE:
+                devs += bond.get_slaves(nmdev)
+
+            ovs_port_dev = ovs.get_port_by_slave(nmdev)
+            if ovs_port_dev:
+                devs.append(ovs_port_dev)
+        return devs
+
+    def _preapply_dns_fix(self):
+        """
+         * When DNS configuration does not changed and old interface hold DNS
+           configuration is not included in `ifaces_desired_state`, preserve
+           the old DNS configure by removing DNS metadata from
+           `ifaces_desired_state`.
+         * When DNS configuration changed, include old interface which is holding
+           DNS configuration, so it's DNS configure could be removed.
+        """
+        cur_dns_iface_names = get_dns_config_iface_names(
+            ipv4.acs_and_ip_profiles(self._ctx.client),
+            ipv6.acs_and_ip_profiles(self._ctx.client),
+        )
+
+        # Whether to mark interface as changed which is used for holding old DNS
+        # configurations
+        remove_existing_dns_config = False
+        # Whether to preserve old DNS config by DNS metadata to be removed from
+        # desired state
+        preserve_old_dns_config = False
+        if self._net_state.dns.config == self._net_state.dns.current_config:
+            for cur_dns_iface_name in cur_dns_iface_names:
+                iface = self._net_state.ifaces[cur_dns_iface_name]
+                if iface.is_changed or iface.is_desired:
+                    remove_existing_dns_config = True
+            if not remove_existing_dns_config:
+                preserve_old_dns_config = True
+        else:
+            remove_existing_dns_config = True
+
+        if remove_existing_dns_config:
+            for cur_dns_iface_name in cur_dns_iface_names:
+                iface = _net_state.ifaces[cur_dns_iface_name]
+                iface.mark_as_changed()
+
+        if preserve_old_dns_config:
+            for iface in self._net_state.ifaces.values():
+                if iface.is_changed or iface.is_desired:
+                    iface.remove_dns_metadata()
+
+
+class Profile:
+    def __init__(self, context, iface=None, profile=None):
```

Comment:
The name profile is blur when the Class is named as `Profile`. Try `nm_remote_conn` or `remote_conn`.

I use `iface_info` for dict iface state, `iface_state` for `BaseIface` object. Maybe `iface_state` here and correct the
`iface_state` used in this file.